### PR TITLE
Add windows tray example

### PR DESCRIPTION
**Pull request by ilopX**
# Add windows tray example

image

## New Win32api functions:
* Shell_NotifyIcon
* GetSubMenu
* DestroyMenu
* TrackPopupMenuEx

## New Struct:
* NOTIFYICONDATA
* TPMPARAMS

## New constant:
* Shell_NotifyIcon uFlags
* Shell_NotifyIcon WndProc callback msg
* NOTIFYICONDATA::uFlags
* NOTIFYICONDATA::dwState
* NOTIFYICONDATA::dwInfoFlags
* TrackPopupMenuEx: uFlag
* LoadImage: type, fuLoad

### CODE REVIEW

**Comment by timsneath** [comment]
Oooh -- this is cool! Thank you. I'll take a look over it. We're just doing some nasty breaking changes in FFI right now, so that's occupying my attention, but this is perfect!

File: lib/src/structs.dart [hunk 1]
```diff
@@ -3070,106 +3364,408 @@ class ACTCTX extends Struct {
...
+      cast<Uint8>().elementAt(824).cast<Utf16>().unpackString(64);
+
+  set szInfoTitle(String text63chars) {
+    if (text63chars.length > 63) {
+      throw ArgumentError('Argument "text63chars" should be less 63 char');
+    }
+    _wchar_cpy(this, 824, text63chars);
+  }
+
+  static void _wchar_cpy(Pointer pointer, int memStart, String source) {
```
Comment (timsneath) [inline lib/src/structs.dart]: This is a neat approach. But there's a similar one in extensions/set_string.dart, and I wonder if we should merge them together. We won't want a static method like this in every extension method that sets strings, I don't think.

File: lib/src/structs.dart [hunk 2]
```diff
@@ -3070,106 +3364,408 @@ class ACTCTX extends Struct {
...
+  external int uFlags;
+
   @Uint32()
-  external int dwReserved0;
+  external int uCallbackMessage;
+
+  @IntPtr()
+  external int hIcon;
+
+  // WCHAR szTip[128]
```
Comment (timsneath) [inline lib/src/structs.dart:3527]: Sorry about all this crud. Inline arrays are coming real soon now (TM): https://github.com/dart-lang/sdk/issues/35763

**Comment by timsneath** [comment]
```diff
diff --git a/lib/src/structs.dart b/lib/src/structs.dart
index 91553f68..a5057eca 100644
--- a/lib/src/structs.dart
+++ b/lib/src/structs.dart
@@ -3699,14 +3699,15 @@ class NOTIFYICONDATA extends Struct {
   @Uint32()
   external int dwInfoFlags;
 
-  @Uint32()
-  external int guidItem1;
-  @Uint16()
-  external int guidItem2;
-  @Uint16()
-  external int guidItem3;
-  @Uint64()
-  external int guidItem4;
+  external GUID guidItem;
+  // @Uint32()
+  // external int guidItem1;
+  // @Uint16()
+  // external int guidItem2;
+  // @Uint16()
+  // external int guidItem3;
+  // @Uint64()
+  // external int guidItem4;
 
   @IntPtr()
   external int hBalloonIcon;

```

**Comment by timsneath** [comment]
```diff
diff --git a/lib/src/structs.dart b/lib/src/structs.dart
index 91553f68..a5057eca 100644
--- a/lib/src/structs.dart
+++ b/lib/src/structs.dart
@@ -3699,14 +3699,15 @@ class NOTIFYICONDATA extends Struct {
   @Uint32()
   external int dwInfoFlags;
 
-  @Uint32()
-  external int guidItem1;
-  @Uint16()
-  external int guidItem2;
-  @Uint16()
-  external int guidItem3;
-  @Uint64()
-  external int guidItem4;
+  external GUID guidItem;
+  // @Uint32()
+  // external int guidItem1;
+  // @Uint16()
+  // external int guidItem2;
+  // @Uint16()
+  // external int guidItem3;
+  // @Uint64()
+  // external int guidItem4;
 
   @IntPtr()
   external int hBalloonIcon;

```

**Comment by timsneath** [comment]
```diff
diff --git a/lib/src/structs.dart b/lib/src/structs.dart
index 91553f68..e42cbc5a 100644
--- a/lib/src/structs.dart
+++ b/lib/src/structs.dart
@@ -3699,14 +3699,7 @@ class NOTIFYICONDATA extends Struct {
   @Uint32()
   external int dwInfoFlags;
 
-  @Uint32()
-  external int guidItem1;
-  @Uint16()
-  external int guidItem2;
-  @Uint16()
-  external int guidItem3;
-  @Uint64()
-  external int guidItem4;
+  external GUID guidItem;
 
   @IntPtr()
   external int hBalloonIcon;

```

**Review by timsneath — CHANGES_REQUESTED** [review]
This is really nice work -- congrats @ilopX. This is not easy to get right, as I've learned through many hours of trial and error. 

If we can resolve the string item for consistency, I'd be delighted to merge this. Note that I'm using the `no-addressof` branch as the development branch while I wait for .224 to land in in the dev branch (so CI is clean). So, unless that has merged at the time  when you resubmit, it would be great if you can target this at that branch and I'll merge there and then land it all into `master` at the same time. Does that work?

**Review by timsneath — APPROVED** [review]
Thank you so much! It's really encouraging to see others contribute high-quality code to this repo. I greatly appreciate you taking the time not only to submit a pull request, but to follow along with the changes for FFI and API metadata!

**Comment by ilopX** [comment]
Thank you. I'm honored that you accepted my pull request. It inspires me a lot.

**Pull request merged**